packages/provider/src/HocuspocusProvider.ts

```diff
@@ -108,6 +108,10 @@ export interface CompleteHocuspocusProviderConfiguration {
   preserveConnection: boolean,
 }
 
+export class AwarenessError extends Error {
+  code = 1001
```

Comment:
An arbitrary error code that is unique across the app. Even though errors can be tested with `err instanceof AwarenessError`, it's a good idea to have a serializable error code that can be logged, referenced in documentation, and compared across library versions.